### ISSUE TITLE
enhancement: watch multiple files; trigger bundling of entry point at change

### ISSUE THREAD

**Issue by adam-devel** [evanw/esbuild#759]
watch mode is added so we don't have to run the same command over and over (and to make bundling faster i guess).

the problem is that you can only watch the entry point file(s) for changes, changes in any other file(s) of the source code does not trigger the bundler

watch mode seem to only be useful when bundling files "one to one", rather than bundling multiple files into a single **bundle**

a bundler is usually used for **bundling** multiple modules scattered around multiple files, into a single file (or at least that's how i am using it), 

for example if `x.ts` is an entry point; it import modules from `a.ts`, `b.ts`, and `c.ts`; `y.js`is the output file.
i think it would make sense to allow for watching `x.ts`, `a.ts`, `b.ts`, and `c.ts` and only bundling `x.ts` at change (even if `x.ts` did not change)

in other words, make changes to the files required by `x.ts` trigger the bundler

**Comment by evanw** [comment]
That's the way watch mode already works. Watch mode watches all files, not just the entry point files. Have you tried editing a non entry point file with watch mode active to verify that it doesn't work? If so, can you say more about your setup (e.g. operating system, package manager, whether you're using JS/Go/CLI, etc.)?

**Comment by adam-devel** [comment]
oh i am really sorry, i rushed in conclusions, just because there was no output in the terminal, i thought it's not reacting to changes, so i assumed that at least, the bundler is going to react to changes in the entry point, this was a problem with another tool i have used before, so i thought esbuild also have that issue, again sorry, closing
